### DISCUSSION TITLE
Is it necessary to install with notebook?

### DISCUSSION THREAD

**Discussion by lyingTree** [pyTMD/pyTMD#94] [General]
I actually think the pyTMD is the best interface to read and load tidal solutions in python.

But it's too large for me, because I only use it to read and load tidal solutions. 
Then I don't think the pyTMD should be installed with the notebook. 
Most of the packages provide relevant jupyter scripts that are just example scripts, not components of the package. It should be selected by the user.

Thank you!

**Comment by tsutterley** [comment]
Hi @lyingTree, the [`requirements.txt`](https://github.com/tsutterley/pyTMD/blob/main/requirements.txt) file contains the more slimmed down requirements. This file would be used when installing with `pip`. I think it only has the packages necessary to compile everything importable. I think it's still a little bloated because I have read/write programs that use HDF5/geotiff formats. I can think about trying to slim things down more. I've also thought about moving some pieces of code out of this repository (mentioned in #72). I am open to suggestions.

**Comment by lyingTree** [comment]
I am very glad to see your reply. Your reply is very useful.
I hope to continue to submit some issues and pr.
